shell/src/main/java/com/sequenceiq/cloudbreak/shell/commands/StackCommands.java

```diff
@@ -297,17 +297,21 @@ private void prepareCluster(String stackId) {
     @CliCommand(value = "stack terminate", help = "Terminate the stack by its id")
     public String terminateStack(
             @CliOption(key = "id", mandatory = false, help = "Id of the stack") String id,
-            @CliOption(key = "name", mandatory = false, help = "Name of the stack") String name) {
+            @CliOption(key = "name", mandatory = false, help = "Name of the stack") String name,
+            @CliOption(key = "wait", mandatory = false, help = "Wait for stack creation", specifiedDefaultValue = "false") Boolean wait) {
         try {
+            wait = wait == null ? false : wait;
             if (id != null) {
                 cloudbreakClient.stackEndpoint().delete(Long.valueOf(id), false);
                 context.setHint(Hints.CREATE_CLUSTER);
                 context.removeStack(id);
-                return "Stack terminated with id: " + id;
+                return wait ? cloudbreakUtil.waitAndCheckStackStatus(Long.valueOf(id), Status.DELETE_COMPLETED.name()) : "Stack terminated with id: " + id;
             } else if (name != null) {
+                StackResponse response = cloudbreakClient.stackEndpoint().getPublic(name);
                 cloudbreakClient.stackEndpoint().deletePublic(name, false);
                 context.setHint(Hints.CREATE_CLUSTER);
-                return "Stack terminated with name: " + name;
+
+                return wait ? cloudbreakUtil.waitAndCheckStackStatus(response.getId(), Status.DELETE_COMPLETED.name()) : "Stack terminated with name: " + name;
```

Comment:
If wait is true then the result of the function is either "FAILED" or "FINISHED", but without waiting the result is "Stack terminated with id: " + id. I should be more consistent.